src/android/CameraLauncher.java

```diff
@@ -778,7 +778,30 @@ private void processResultFromGallery(int destType, Intent intent) {
 
                     // If sending base64 image back
                     if (destType == DATA_URL) {
-                        this.processPicture(bitmap, this.encodingType);
+                        // To allow keeping the Exif data, we need to save the image to allow ExifInterface add the data
+                        // After that, we load the image and send it as base64
+                        try {
+                            String modifiedPath = this.outputModifiedBitmap(bitmap, uri, mimeTypeOfGalleryFile);
+                            InputStream fileStream = FileHelper.getInputStreamFromUriString(modifiedPath, cordova);
+                            ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+
+                            int nRead;
+                            byte[] data = new byte[512];
```

Comment:
Looks good, I enabled the unit tests but I think we probably want to use a larger buffer than `512` bytes here.

default memory limits of android apps are `256mb` and this is used for images which is probably on average 5-10mb depending on the resolution of the captured image.

My Galaxy Note 9 takes pictures and it ranges from 3-5mb in JPG format.

```suggestion
                            final int MB = 1048576;
                            byte[] data = new byte[MB * 4];
```

Some phones may have larger images, other phones may have smaller images, but here we have a 4mb buffer which isn't too much compared to the 256mb memory limit of the app, and devices today aren't so memory constrained that a 4mb buffer will matter. The 4mb buffer will likely be large enough to fit an entire image, which is the closest behaviour that you had previously when you were trying to use `.avaiable()`, allowing the file to be read copied in a single loop pass.

If the image happens to be larger, then we drastically reduce the amount of times it needs to reset and fill the buffer. (e.g. if the image was 5mb, then we would need to do `10240` iterations to read the entire image with a 512 byte buffer, which would be kinda slow).

The `MB` constant is used here purely for readability so we know the value is 1 megabyte.